cache.go

```diff
@@ -91,11 +96,26 @@ func (c *cache) get(qname string) RRs {
 	if len(e) == 0 {
 		return emptyRRs
 	}
-	i := 0
-	rrs := make(RRs, len(e))
-	for rr, _ := range e {
-		rrs[i] = rr
-		i++
+	if c.expire {
+		i := 0
+		rrs := make(RRs, len(e))
+		now := time.Now()
```

Comment:
Grab a single `time.Now()` at the start of this function.

Reply:
I think this one is OK. It happens outside the loop and only in the branch where expiration is enabled, so it's not creating times when they won't be used.